packages/appsync-modelgen-plugin/src/__tests__/visitors/__snapshots__/appsync-swift-visitor.test.ts.snap

```diff
@@ -7,6 +7,256 @@ import Foundation
+      teamId: teamId,
+      teamName: teamName,
+      createdAt: nil,
+      updatedAt: nil)
+  }
+  internal init(projectId: String,
+      name: String,
+      team: Team? = nil,
+      teamId: String? = nil,
+      teamName: String? = nil,
+      createdAt: Temporal.DateTime? = nil,
+      updatedAt: Temporal.DateTime? = nil) {
+      self.projectId = projectId
+      self.name = name
+      self._team = LazyReference(team)
+      self.teamId = teamId
+      self.teamName = teamName
+      self.createdAt = createdAt
+      self.updatedAt = updatedAt
+  }
+  public mutating func setTeam(_ team: Team? = nil) {
+    self._team = LazyReference(team)
+  }
+  public init(from decoder: Decoder) throws {
+      let values = try decoder.container(keyedBy: CodingKeys.self)
+      projectId = try values.decode(String.self, forKey: .projectId)
+      name = try values.decode(String.self, forKey: .name)
+      _team = try values.decodeIfPresent(LazyReference<Team>.self, forKey: .team) ?? LazyReference(identifiers: nil)
+      teamId = try? values.decode(String?.self, forKey: .teamId)
+      teamName = try? values.decode(String?.self, forKey: .teamName)
+      createdAt = try? values.decode(Temporal.DateTime?.self, forKey: .createdAt)
+      updatedAt = try? values.decode(Temporal.DateTime?.self, forKey: .updatedAt)
+  }
+  public func encode(to encoder: Encoder) throws {
+      var container = encoder.container(keyedBy: CodingKeys.self)
+      try container.encode(projectId, forKey: .projectId)
+      try container.encode(name, forKey: .name)
+      try container.encode(_team, forKey: .team)
+      try container.encode(teamId, forKey: .teamId)
+      try container.encode(teamName, forKey: .teamName)
+      try container.encode(createdAt, forKey: .createdAt)
+      try container.encode(updatedAt, forKey: .updatedAt)
+  }
+}"
+`;
+
+exports[`AppSyncSwiftVisitor Primary Key Tests Should generate associatedFields in hasOne uni-directional relation for models with a composite PK 2`] = `
+"// swiftlint:disable all
+import Amplify
+import Foundation
+
+extension Project {
+  // MARK: - CodingKeys 
+   public enum CodingKeys: String, ModelKey {
+    case projectId
+    case name
+    case team
+    case teamId
+    case teamName
+    case createdAt
+    case updatedAt
+  }
+  
+  public static let keys = CodingKeys.self
+  //  MARK: - ModelSchema 
+  
+  public static let schema = defineSchema { model in
+    let project = Project.keys
+    
+    model.pluralName = \\"Projects\\"
+    
+    model.attributes(
+      .index(fields: [\\"projectId\\", \\"name\\"], name: nil),
+      .primaryKey(fields: [project.projectId, project.name])
+    )
+    
+    model.fields(
+      .field(project.projectId, is: .required, ofType: .string),
+      .field(project.name, is: .required, ofType: .string),
+      .hasOne(project.team, is: .optional, ofType: Team.self, associatedFields: [Team.keys.teamId, Team.keys.name], targetNames: [\\"teamId\\", \\"teamName\\"]),
```

Comment:
Because we are introducing `associatedFields`, we need to add this definition to the library first.